File: apps/chat/libs/chat-domain/src/main/java/com/example/chat/domain/channel/ChannelId.java
package com.example.chat.domain.channel;

import java.util.UUID;

/**
 * 채널 ID (Value Object)
 */
public record ChannelId(String value) {
    public ChannelId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("ChannelId cannot be null or blank");
        }
    }

    public static ChannelId of(String value) {
        return new ChannelId(value);
    }

    public static ChannelId generate() {
        return new ChannelId(UUID.randomUUID().toString());
    }

    public String getValue() {
        return value();
    }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/channel/ChannelId.kt
package com.example.chat.domain.channel

import java.util.UUID

/**
 * 채널 ID (Value Object)
 */
data class ChannelId(
    val value: String
) {
    init {
        require(value.isNotBlank()) { "ChannelId cannot be null or blank" }
    }

    companion object {
        fun of(value: String): ChannelId = ChannelId(value)

        fun generate(): ChannelId = ChannelId(UUID.randomUUID().toString())
    }
}

